pkg/actuators/machine/instances.go

```diff
@@ -363,11 +367,14 @@ func launchInstance(machine *machinev1beta1.Machine, machineProviderConfig *mach
 	subnetID := subnetIDs[0]
 	var networkInterfaces = []*ec2.InstanceNetworkInterfaceSpecification{
 		{
-			DeviceIndex: aws.Int64(machineProviderConfig.DeviceIndex),
-			SubnetId:    subnetID,
-			Groups:      securityGroupsIDs,
+			DeviceIndex:               aws.Int64(machineProviderConfig.DeviceIndex),
+			SubnetId:                  subnetID,
+			Groups:                    securityGroupsIDs,
+			AssociatePublicIpAddress:  aws.Bool(false),
+			AssociateCarrierIpAddress: aws.Bool(false),
```

Comment:
I would rather we passthrough the pointer and let AWS decide whether a `nil` value has meaning or not, this would be in-keeping with what the upstream CAPA implementation does

Reply:
Okay, @JoelSpeed . Thanks. Let me know if this makes sense now. I am following your suggestion, considering the mutually exclusive fields when using Waevlength infra.